### PR TITLE
task import: handle errors during import better

### LINKED ISSUE #1091: Improve error handling for invalid ProFormA files
Currently, we are just showing a very generic Internal Server Error whenever a ProFormA file could not be processed correctly:

ProFormA error

However, it would be great for users to learn more about the error, so that it can be fixed by themselves (without throwing an exception in the backend). This ticket is designed to improve the error handling in the backend (no exception, do not respond with 503, ...) and frontend (proper error message, do not display `null`, ...).

Here is an invalid example: [task 2.0.1.zip](https://github.com/openHPI/codeharbor/files/13390466/task.2.0.1.zip)

### PR DESCRIPTION
**Pull request by kkoehn**
fixes #1091

removed unnecessary alerts and added more rescues

### CODE REVIEW

File: config/locales/de/controllers/tasks.yml [hunk 1]
```diff
@@ -9,15 +9,17 @@ de:
     export_external_confirm:
       error: 'Der Export der Aufgabe (%{title}) ist fehlgeschlagen. <br> Fehler: %{error}'
       success: Aufgabe (%{title}) erfolgreich exportiert.
+    import:
+      internal_error: Bei der Importierung der Aufgabe ist ein interner Fehler aufgetreten.
```
Comment (MrSerth) [inline config/locales/de/controllers/tasks.yml:13]: Just trying to improve the German wording a little; I know that you just moved it (due to the changed key):

```suggestion
      internal_error: Beim Import dieser Aufgabe ist auf CodeHarbor ein interner Fehler aufgetreten.
```

File: config/locales/en/proforma_errors.yml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+---
+en:
+  proforma_errors:
+    version not supported: The version of this ProformaXML document is not supported
```
Comment (MrSerth) [inline config/locales/en/proforma_errors.yml:4]:
```suggestion
    version not supported: The version of this ProformaXML document is not supported.
```

**Review by MrSerth — CHANGES_REQUESTED** [review]
Nice, thank you! I really like the new error handling and think this will improve the fault tolerance. I have a few minor suggestions, but also one larger one regarding the current error handling: At least in my tests, I was able to generate another exception in the existing error handling (rescuing `ProformaXML::ProformaError`). I annotated two examples and hope that this will allow you to make the handling even more robust.

File: config/locales/de/proforma_errors.yml [hunk 3]
```diff
@@ -1,4 +1,7 @@
 ---
 de:
   proforma_errors:
-    version not supported: Die Version dieses ProformaXML Dokuments wird nicht unterstützt
+    import: Es ist ein Fehler beim Importieren der ProformaXML Zip-Datei aufgetreten.
+    import_default: "%{error}"
+    no task_xml found: Es konnte keine task.xml in der Zip-Datei gefunden werden.
+    version not supported: Die Version dieses ProformaXML Dokuments wird nicht unterstützt.
```
Comment (MrSerth) [inline config/locales/de/proforma_errors.yml:7]: We recently aimed to unify the spelling to `ZIP-Datei`.

File: config/locales/en/proforma_errors.yml [hunk 4]
```diff
@@ -1,4 +1,7 @@
 ---
 en:
   proforma_errors:
-    version not supported: The version of this ProformaXML document is not supported
+    import: An error occurred while importing your ProformaXML zip-file.
+    import_default: "%{error}"
+    no task_xml found: No task.xml could be found in the supplied zip-file.
+    version not supported: The version of this ProformaXML document is not supported.
```
Comment (MrSerth) [inline config/locales/en/proforma_errors.yml:7]: In English, we use `ZIP file`.

**Review by MrSerth — APPROVED** [review]
Thanks! All errors I've tested were correctly handled and it looks nice. Thanks!

File: config/locales/de/proforma_errors.yml [hunk 5]
```diff
@@ -0,0 +1,7 @@
+---
+de:
+  proforma_errors:
+    import: Es ist ein Fehler beim Importieren der ProformaXML ZIP-Datei aufgetreten.
+    import_default: "%{error}"
```
Comment (Dome-GER) [inline config/locales/de/proforma_errors.yml:5]: What is the benefit of this locale (in contrast to passing in the `msg` directly to the `default`)?

File: spec/controllers/tasks_controller_spec.rb [hunk 6]
```diff
@@ -736,6 +736,28 @@
         expect(JSON.parse(response.body, symbolize_names: true)).to eql({status: 'failure', message: 'You need to choose a file.'})
       end
     end
+
+    context 'when service throws version not supported error' do
+      before do
+        allow(ProformaService::CacheImportFile).to receive(:call).and_raise(ProformaXML::ProformaError.new(['version not supported']))
+      end
+
+      it 'renders correct json' do
```
Comment (Dome-GER) [inline spec/controllers/tasks_controller_spec.rb:745]:
```suggestion
      it 'renders correct JSON' do
```

Nitpick. JSON is an acronym. But you probably use `json` everywhere.

**Review by Dome-GER — APPROVED** [review]
The `prettify_import_errors` looks a bit brittle. But I also don't have a much better idea right now.

**Pull request merged**